internal/runtime/client/client.go

```diff
@@ -524,6 +529,10 @@ func httpCall(ctx context.Context, request, response runtime.Object, opts *httpC
 		}
 	}
 
+	if responseObject, ok := response.(runtimehooksv1.ResponseObject); ok {
+		respStatus = string(responseObject.GetStatus())
+	}
+
```

Comment:
As this is only required for the metric: let's move this into the observe function so we also get rid of using defer?

WDYT?

Reply:
thanks! This typecast will be more suitable in the `Observe` function. But I still don't understand how to get rid of `defer` func. 

e.g. if we will make `Observe` function to be like:
```
// Observe observes a http request result and increments the metric for the given
// http status code, host, gvh and response status.
func (m *requestsTotalObserver) Observe(req *http.Request, resp *http.Response, gvh runtimecatalog.GroupVersionHook, err error, response runtime.Object) {
	host := req.URL.Host

	// Errors can be arbitrary strings. Unbound label cardinality is not suitable for a metric
	// system so they are reported as `<error>`.
	code := "<error>"
	if err == nil {
		code = strconv.Itoa(resp.StatusCode)
	}

	status := unknownResponseStatus
	if responseObject, ok := response.(runtimehooksv1.ResponseObject); ok && responseObject.GetStatus() != "" {
		status = string(responseObject.GetStatus())
	}

	m.metric.WithLabelValues(code, host, gvh.Group, gvh.Version, gvh.Hook, status).Inc()
}
```

we will still need to use `defer` because there are a lot `return` code paths in `httpCall` function before the `response` variable is actually set.

Reply:
Ah I see. Did somehow not see that the body is read on a later stage in the func. :+1:

Feel free to resolve here.